de.fhdo.lemma.service.open_api/src/de/fhdo/lemma/service/open_api/LemmaTechnologySubGenerator.xtend

```diff
@@ -0,0 +1,181 @@
+        //Sets "application/json" as default format if the OpenAPI desc entails it at all.
+        // If not, the first encountered media type is set as default.
+        rest.defaultFormat = rest.dataFormats.findFirst[
+            it.formatName.equals("application/json")
+        ] ?: rest.dataFormats.get(0)
+        myTechModel.protocols.add(rest)
+    }
+
+    def searchMediaTimes(PathItem item) {
+        val httpVerbList = <Operation>newArrayList
+        httpVerbList.addAll(item.get, item.delete, item.put,
+            item.post, item.patch, item.head, item.options
+        )
+        httpVerbList.forEach [ verb |
+            if (verb !== null) {
+                // Getting MediaTypes from responses
+                verb.responses?.forEach[
+                    responseCode, responseSchema|addToMediaTypes(responseSchema?.content?.keySet)
+                ]
+                // Getting MediaTypes from requests
+                addToMediaTypes(verb.requestBody?.content?.keySet)
+            }
+
+        ]
+    }
+
+    def addToMediaTypes(Set<String> strings) {
+        if (strings !== null) {
+            strings.forEach[mediaType|mediaTypes.add(mediaType)]
+        }
+    }
+
+    def initilize() {
+        myTechModel.name = OpenApiUtil.removeInvalidCharsFromName("OpenApi")
+        // Adding Primitive Types for OpenApi Technology
+        myTechModel.initPrimitives
+        myTechModel.initAspects
+        // TODO ResponseCodes, Operation Types, Formats
+    }
+
+    /** Creates Default Aspects for HTTP-based communication,
+     * i.e., request methods and response codes.
+     */
+    def void initAspects(Technology technology) {
+        val httpVerbs = <ServiceAspect>newArrayList
+        httpVerbs.add("GetMapping".createServiceAspect)
+        httpVerbs.add("PutMapping".createServiceAspect)
+        httpVerbs.add("PostMapping".createServiceAspect)
+        httpVerbs.add("DeleteMapping".createServiceAspect)
+        httpVerbs.add("OptionsMapping".createServiceAspect)
+        httpVerbs.add("HeadMapping".createServiceAspect)
+        httpVerbs.add("PatchMapping".createServiceAspect)
+        httpVerbs.add("TraceMapping".createServiceAspect)
+        httpVerbs.add("Unspecified".createServiceAspect)
+        myTechModel.serviceAspects.addAll(httpVerbs)
+    }
+
+    def ServiceAspect createServiceAspect(String name) {
+        val ret = TECHNOLOGY_FACTORY.createServiceAspect
+        ret.name = name
+        ret.joinPoints.add(JoinPointType.OPERATIONS)
+        ret.features.add(AspectFeature.SINGLE_VALUED)
+        return ret
+    }
+
+    /** Creates OpenApi data types corresponding to <italic>OpenApi v3.0.3 - Data Types</italic>.
+     * OpenApi data type formats are represented as individual types.
+     */
+    def void initPrimitives(Technology technology) {
+        val primTypes = <TechnologySpecificPrimitiveType>newArrayList
+
+        primTypes.add(createPrimitive("Integer", true, DATA_FACTORY.createPrimitiveInteger))
```

Comment:
See my comments above. Why not making `primtTypes` a class attribute being initialized exactly once per instance?